miniz_tinfl.c

```diff
@@ -203,6 +203,10 @@ tinfl_status tinfl_decompress(tinfl_decompressor *r, const mz_uint8 *pIn_buf_nex
     size_t out_buf_size_mask = (decomp_flags & TINFL_FLAG_USING_NON_WRAPPING_OUTPUT_BUF) ? (size_t)-1 : ((pOut_buf_next - pOut_buf_start) + *pOut_buf_size) - 1, dist_from_out_buf_start;
 
     /* Ensure the output buffer's size is a power of 2, unless the output buffer is large enough to hold the entire output file (in which case it doesn't matter). */
+    if ((!pOut_buf_start) || (!pOut_buf_next) || (!pIn_buf_size) || (!pOut_buf_size))
```

Comment:
This might impact performance but YMMV

Reply:
This part of the patch is wrong: `pOut_buf_next` can legitimately be NULL here, and this change results in decompression failure with valid files.

Reply:
I'm hitting `pOut_buf_start` == `NULL` with `tinfl_decompress` called from `tinfl_decompress_mem_to_heap`

Reply:
@uroni: I suggest reverting commit fe6330e1d8b975ff075ab3bd85709480727e8afa

Reply:
I see that in `tinfl_decompress_mem_to_heap`, the first iteration of the main `for` loop has `pBuf` set to `NULL` and the resulting call to `tinfl_decompress` will have both `pOut_buf_start` and `pOut_buf_next` set to `NULL`. `tinfl_decompress` responds to this with `TINFL_STATUS_HAS_MORE_OUTPUT` if the check is eliminated, which is correct behaviour. Calling this function with either buffer being `NULL` while the other is valid also results in a valid error code. Calling it with either buffer length pointer being `NULL` results in a read access violation **before** the test. I therefore think it would be best to:

1) Remove the test for `NULL` for `pOut_buf_start` and `pOut_buf_next`, and,
2) Rework the initialization to test for `NULL` for `pIn_buf_size` and `pOut_buf_size`.